src/nvim/buffer_updates.c

```diff
@@ -0,0 +1,214 @@
+#include "nvim/buffer_updates.h"
+#include "nvim/memline.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/msgpack_rpc/channel.h"
+#include "nvim/assert.h"
+
+// Register a channel. Return True if the channel was added, or already added.
+// Return False if the channel couldn't be added because the buffer is
+// unloaded.
+bool buffer_updates_register(buf_T *buf, uint64_t channel_id, bool send_buffer)
```

Comment:
file names are buffer.c/buffer_updates.c but all functions always start with `buf_`, not `buffer_`. Let's stay consistent on this.